src/Shouldly/ShouldlyExtensionMethods/ShouldBe/GenericShouldBeTestExtensions.cs

```diff
@@ -33,9 +37,9 @@ public static void ShouldNotBe<T>([AllowNull] this T actual, [AllowNull] T expec
         }
 
         [ContractAnnotation("actual:null,expected:null => halt")]
-        public static void ShouldNotBe<T>(this T actual, T expected, IEqualityComparer<T> comparer, string? customMessage = null)
+        public static void ShouldNotBe<T>([AllowNull] this T actual, [AllowNull] T expected, IEqualityComparer<T> comparer, string? customMessage = null)
         {
-            actual.AssertAwesomely(v => !Is.Equal(v, expected, comparer), actual, expected, customMessage);
+            actual!.AssertAwesomely(v => !Is.Equal(v, expected, comparer), actual, expected, customMessage);
```

Comment:
Not sure why AssertAwesomely is annotated such that `actual` requires `!`, but that's outside the scope of this PR.